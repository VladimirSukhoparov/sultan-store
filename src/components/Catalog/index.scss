.catalog {
    
    &__link {
        
        text-decoration: none;
        text-align: center;
        display: inline-block;

        &:hover {
            text-decoration: underline;
        }
        figure{
            box-sizing: border-box;
            padding: 5px;
            margin: 20px 0;
            img{
                width: 250px;
                border-radius: 20px;
                background:#fff6e4;
    
            }
        }
    }

    &__title {
        color: black;
        font-weight: 500;
        font-size: 16px;
        line-height: 130%;
        margin-top: 10px;
        

    }
}