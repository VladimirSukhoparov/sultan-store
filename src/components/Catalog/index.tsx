import React from "react";
import { NavLink } from "react-router-dom";
import cosmetics from "../../images/cosmetics.png";

import "./index.scss";

export const Catalog = () => {
  return (
    <div className="catalog" data-testid="catalog">
      <NavLink to="cosmetics" className="catalog__link">
        <figure data-testid="fugure">
          <img src={cosmetics} alt="cosmetics" />
          <figcaption className="catalog__title">
            Косметика и гигиена
          </figcaption>
        </figure>
      </NavLink>
    </div>
  );
};
