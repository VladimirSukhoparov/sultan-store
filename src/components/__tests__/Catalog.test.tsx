import { screen, render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import { Catalog } from "../Catalog";

afterEach(() => {
  cleanup();
});

test("should render catalog component", () => {
  render(
    <BrowserRouter>
      <Catalog />
    </BrowserRouter>
  );
  const catalogElement = screen.getByTestId("catalog");
  expect(catalogElement).toBeInTheDocument();
  const fugureElement = screen.getByTestId("fugure");
  expect(fugureElement).toContainHTML(
    "<img src='cosmetics.png' alt='cosmetics'/><figcaption class='catalog__title'>Косметика и гигиена</figcaption>"
  );
});

test("matches snapshot", () => {
  const catalog = renderer
    .create(
      <BrowserRouter>
        <Catalog />
      </BrowserRouter>
    )
    .toJSON();
  expect(catalog).toMatchSnapshot();
});
