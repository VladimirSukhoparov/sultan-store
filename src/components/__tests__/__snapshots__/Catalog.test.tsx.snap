// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="catalog"
  data-testid="catalog"
>
  <a
    className="catalog__link"
    href="/cosmetics"
    onClick={[Function]}
  >
    <figure
      data-testid="fugure"
    >
      <img
        alt="cosmetics"
        src="cosmetics.png"
      />
      <figcaption
        className="catalog__title"
      >
        Косметика и гигиена
      </figcaption>
    </figure>
  </a>
</div>
`;
